### ISSUE TITLE
Windows support

### ISSUE THREAD

**Issue by krille-chan** [krille-chan/fluffychat#162]
#### Description

I know FluffyChat has a Windows build (somewhere in GitLab or build artifacts), but I understand that it doesn't have automatic updating and isn't officially supported yet as it's not listed on FluffyChat.im. I am not able to find a tracking issue though.

* https://fluffychat.im/

I am aware of the PWA, but considering I clear everything on browser close and may close browser more often than I log out (#596), I would prefer using desktop app without doing anything unsupported.

**Comment by Doko-Demo-Doa** [comment]
So... is the Windows build working yet? I'm just wondering.

**Comment by TLCPlusPlus** [comment]
> So... is the Windows build working yet? I'm just wondering.

No. It would be in the Releases or at the very least, the website mentioned.

**Comment by compuguy** [comment]
I've recently been trying to go through the process of building the latest release for windows. It seems to be failing to build the `flutter_secure_storage_windows` plugin:

```
[   +2 ms] Done Building Project
"C:\Users\user\git\fluffychat\build\windows\x64\plugins\window_to_front\window_to_front_plugin.vcxproj" (default
targets).
[   +2 ms] Done Building Project "C:\Users\user\git\fluffychat\build\windows\x64\runner\fluffychat.vcxproj" (default
targets) -- FAILED.
[        ] Done Building Project "C:\Users\user\git\fluffychat\build\windows\x64\ALL_BUILD.vcxproj" (default targets) --
FAILED.
[        ] Done Building Project "C:\Users\user\git\fluffychat\build\windows\x64\INSTALL.vcxproj" (default targets) --
FAILED.
[  +11 ms] Build FAILED.
[   +3 ms] "C:\Users\user\git\fluffychat\build\windows\x64\INSTALL.vcxproj" (default target) (1) ->
[        ] "C:\Users\user\git\fluffychat\build\windows\x64\ALL_BUILD.vcxproj" (default target) (3) ->
[        ] "C:\Users\user\git\fluffychat\build\windows\x64\runner\fluffychat.vcxproj" (default target) (10) ->
[        ]
"C:\Users\user\git\fluffychat\build\windows\x64\plugins\flutter_secure_storage_windows\flutter_secure_storage_windows_pl
ugin.vcxproj" (default target) (11) ->
[        ] (ClCompile target) ->
[        ]
C:\Users\user\git\fluffychat\windows\flutter\ephemeral\.plugin_symlinks\flutter_secure_storage_windows\windows\flutter_s
ecure_storage_windows_plugin.cpp(6,10): error C1083: Cannot open include file: 'atlstr.h': No such file or directory
[C:\Users\user\git\fluffychat\build\windows\x64\plugins\flutter_secure_storage_windows\flutter_secure_storage_windows_pl
ugin.vcxproj]
[        ]     0 Warning(s)
[        ]     1 Error(s)
[        ] Time Elapsed 00:05:34.30
[  +55 ms] Building Windows application... (completed in 360.9s)
[   +4 ms] "flutter windows" took 365,367ms.
[  +10 ms] Build process failed.
[   +1 ms]
           #0      throwToolExit (package:flutter_tools/src/base/common.dart:10:3)
           #1      _runBuild (package:flutter_tools/src/windows/build_windows.dart:253:5)
           <asynchronous suspension>
           #2      buildWindows (package:flutter_tools/src/windows/build_windows.dart:106:5)
           <asynchronous suspension>
           #3      BuildWindowsCommand.runCommand (package:flutter_tools/src/commands/build_windows.dart:56:5)
           <asynchronous suspension>
           #4      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1367:27)
           <asynchronous suspension>
           #5      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #6      CommandRunner.runCommand (package:args/command_runner.dart:212:13)
           <asynchronous suspension>
           #7      FlutterCommandRunner.runCommand.<anonymous closure>
           (package:flutter_tools/src/runner/flutter_command_runner.dart:348:9)
           <asynchronous suspension>
           #8      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #9      FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:294:5)
           <asynchronous suspension>
           #10     run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:112:9)
           <asynchronous suspension>
           #11     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #12     main (package:flutter_tools/executable.dart:90:3)
           <asynchronous suspension>


[  +39 ms] ensureAnalyticsSent: 36ms
[        ] Running 0 shutdown hooks
[        ] Shutdown hooks complete
[        ] exiting with code 1
```

**Comment by CEbbinghaus** [comment]
Is this still a planned feature? Is there a roadmap to capture work that is planned / in progress? I find GitHub Projects work really well for managing that kind of stuff

**Comment by MolotovCherry** [comment]
I'm successfully able to get this to build on Windows\* after making sure build deps are solved (nuget, openssl, nasm). However, upon starting the app, a permanent white screen is displayed which does nothing.

\* note to others: you need to build in admin terminal; the build auto installs itself along with all files to fluffychat dir in program files for some reason. if you go to the flutter build dir, you will only see 1 exe

**Comment by AbyssDayStar** [comment]
I successfully built the Windows version using GitHub Actions, and the basic features work fine.
Before the build, I noticed that WindowsInitializationSettings in background_push.dart and client_manager.dart hadn't been initialized, so I added it.
From the runtime logs, I found the white screen was caused by a missing libsqlcipher.dll. The actual DLL name is sqlite.dll — renaming it to libsqlcipher.dll fixed the issue.
However, the manually built Windows version lacks auto‑updates, a persistent system tray icon, and notification alerts. These are essential features for an IM app, but currently I have to handle them manually.
I can't solve these problems on my own, so I hope the official team can provide support.
